app/src/main/java/com/teamagam/gimelgimel/app/map/actions/timeplay/TimeplayViewModel.java

```diff
@@ -200,23 +243,37 @@ private void showSnapshot(long newTimestamp) {
   }
 
   private void stopPreviousSnapshotDisplay() {
-    if (mSnapshotInteractor != null) {
-      mSnapshotInteractor.unsubscribe();
-    }
+    unsubscribe(mSnapshotInteractor);
   }
 
   private void startCurrentSnapshotDisplay(long newTimestamp) {
-    mSnapshotInteractor =
-        mSnapshotTimeplayInteractorFactory.create(mTimeplayDisplayer, newTimestamp);
+    if (hasNotStarted()) {
+      mSnapshotInteractor =
+          mSnapshotTimeplayInteractorFactory.create(mTimeplayDisplayer, newTimestamp);
+    } else {
+      mSnapshotInteractor =
+          mDatesRangeSnapshotTimeplayInteractorFactory.create(mStartTimestamp, mEndTimestamp,
+              mTimeplayDisplayer, newTimestamp);
+    }
     mSnapshotInteractor.execute();
   }
 
+  public enum PlaySpeed {
+    Low,
+    Medium,
+    High
+  }
+
   interface MapDisplayer {
     void addToMap(GeoEntity geoEntity);
 
     void removeFromMap(GeoEntity geoEntity);
   }
 
+  public interface DialogShower {
```

Comment:
What is this interface good for ?
You're passing its concrete implementation to the view-model and use its specific (not-in-the-interface) methods.
So you dont really make the use of the interface.

**It can be removed from the system and nothing will be effected**
This is the most obvious sign it's useless.
Fewer code lines - fewer code lines to read  - fewer code lines to maintain.

That's not to say interfaces are not important at all, they should be used to achieve IOC (Inversion of control) where it is needed

Reply:
ok